cli/flox-rust-sdk/src/flox.rs

```diff
@@ -276,27 +278,45 @@ impl Floxhub {
     }
 
     pub fn git_url_override(&self) -> Option<&Url> {
-        self.git_url_override.as_ref()
+        self.git_url_overridden.then_some(&self.git_url)
     }
 
     /// Return the url of the floxhub git interface
     ///
     /// If the environment variable `_FLOX_FLOXHUB_GIT_URL` is set,
     /// it will be used instead of the derived floxhub host.
     /// This is useful for testing floxhub locally.
-    pub fn git_url(&self) -> Result<Url, url::ParseError> {
-        if let Some(ref url) = self.git_url_override {
-            return Ok(url.clone());
-        }
-
-        let mut url = self.base_url.clone();
-        let host = url.host_str().unwrap();
-        url.set_host(Some(&format!("git.{}", host)))?;
+    pub fn git_url(&self) -> &Url {
+        &self.git_url
+    }
 
-        Ok(url)
+    fn derive_git_url(base_url: &Url) -> Result<Url, FloxhubError> {
+        let mut git_url = base_url.clone();
+        let host = git_url
+            .host_str()
+            .ok_or(FloxhubError::NoHost(base_url.to_string()))?;
+        let without_hub = host
+            .strip_prefix("hub")
+            .ok_or(FloxhubError::NoHubPrefix(base_url.to_string()))?;
+        let with_git_prefix = format!("api{}", without_hub);
+        git_url
+            .set_host(Some(&with_git_prefix))
+            .map_err(|e| FloxhubError::InvalidFloxhubBaseUrl(with_git_prefix, e))?;
+        git_url.set_path("git");
+        Ok(git_url)
     }
 }
 
+#[derive(Error, Debug)]
+pub enum FloxhubError {
+    #[error("Invalid FloxHub URL: '{0}'. URL must contain a host.")]
+    NoHost(String),
+    #[error("Invalid FloxHub URL: '{0}'. URL must begin with 'hub'")]
+    NoHubPrefix(String),
+    #[error("Couldn't set git URL host to '{0}'")]
+    InvalidFloxhubBaseUrl(String, #[source] url::ParseError),
```

Comment:
suggestion nb: include original url
Technically, this might also be an issue with the rest of the url -- but then again if we were able to extract the host, we can also assume we can set a host.